src/driver/db.ts

```diff
@@ -0,0 +1,190 @@
+        }
+        return this.astraDb.listCollections({ ...options, nameOnly: false });
+    }
+
+    /**
+     * List all tables in the database.
+     */
+    async listTables(options: ListTablesOptions) {
+        if (options.nameOnly) {
+            return this.astraDb.listTables({ ...options, nameOnly: true });
+        }
+        return this.astraDb.listTables({ ...options, nameOnly: false });
+    }
+
+    /**
+     * Execute a command against the database.
+     * @param command The command to be executed.
+     */
+    async command(command: Record<string, unknown>): Promise<RawDataAPIResponse> {
+        return this.astraDb.command(command);
+    }
+}
+
+/**
+ * Db instance that creates and manages collections.
+ * @extends BaseDb
+ */
+export class CollectionsDb extends BaseDb {
+    /**
+     * Creates an instance of CollectionsDb. Do not instantiate this class directly.
+     * @param astraDb The AstraDb instance to interact with the database.
+     * @param keyspaceName The name of the keyspace to use.
+     */
+    constructor(astraDb: AstraDb, keyspaceName: string) {
+        super(astraDb, keyspaceName, false);
+    }
+
+    /**
+     * Get a collection by name.
+     * @param name The name of the collection.
+     */
+    collection<DocType extends Record<string, unknown> = Record<string, unknown>>(name: string, options: CollectionOptions) {
+        return this.astraDb.collection<DocType>(name, options);
+    }
+
+    /**
+     * Send a CreateCollection command to Data API.
+     */
+    createCollection(name: string, options?: Record<string, unknown>) {
+        return this.astraDb.createCollection(name, options);
+    }
+}
+
+/**
+ * Db instance that creates and manages tables.
+ * @extends BaseDb
+ */
+export class TablesDb extends BaseDb {
+    /**
+     * Creates an instance of TablesDb. Do not instantiate this class directly.
+     * @param astraDb The AstraDb instance to interact with the database.
+     * @param keyspaceName The name of the keyspace to use.
+     */
+    constructor(astraDb: AstraDb, keyspaceName: string) {
+        super(astraDb, keyspaceName, true);
+    }
+
+    /**
+     * Get a table by name. This method is called `collection()` for compatibility with Mongoose, which calls
+     * this method for getting a Mongoose Collection instance, which may map to a table in Astra DB when using tables mode.
+     * @param name The name of the table.
+     */
+    collection<DocType extends Record<string, unknown> = Record<string, unknown>>(name: string, options: TableOptions) {
+        return this.astraDb.table<DocType>(name, options);
+    }
+
+    /**
+     * Throws an error, stargate-mongoose does not support creating collections in tables mode.
+     */
+    createCollection(name: string, options?: Record<string, unknown>): Promise<Collection<Record<string, never>>> {
+        throw new StargateMongooseError('Cannot createCollection in tables mode', { name, options });
```

Comment:
```suggestion
        throw new StargateMongooseError('Cannot createCollection in tables mode; use createTable instead', { name, options });
```
Could be a helpful pointer?